core/arch/arm/kernel/spin_lock_a64.S

```diff
@@ -74,10 +74,13 @@ END_FUNC __cpu_spin_lock
 FUNC __cpu_spin_trylock , :
 	mov	w2, #SPINLOCK_LOCK
 .loop:	ldaxr	w1, [x0]
-	cbnz	w1, .cpu_spin_trylock_out
+	cbnz	w1, .failed
 	stxr	w1, w2, [x0]
 	cbnz	w1, .loop
-.cpu_spin_trylock_out:
+	mov	x0, #0
+	ret
+.failed:
+	mov	x0, #1
```

Comment:
i think i'll swap the sign-off and reviewed-by tags :)>  
i update with your proposal.